openmc/mgxs/mgxs.py

```diff
@@ -0,0 +1,5909 @@
+        The keys into the tallies dictionary for each tally used to compute
+        the multi-group cross section
+    estimator : {'tracklength', 'collision', 'analog'}
+        The tally estimator used to compute the multi-group cross section
+    tallies : collections.OrderedDict
+        OpenMC tallies needed to compute the multi-group cross section
+    rxn_rate_tally : openmc.Tally
+        Derived tally for the reaction rate tally used in the numerator to
+        compute the multi-group cross section. This attribute is None
+        unless the multi-group cross section has been computed.
+    xs_tally : openmc.Tally
+        Derived tally for the multi-group cross section. This attribute
+        is None unless the multi-group cross section has been computed.
+    num_subdomains : int
+        The number of subdomains is unity for 'material', 'cell' and 'universe'
+        domain types. This is equal to the number of cell instances
+        for 'distribcell' domain types (it is equal to unity prior to loading
+        tally data from a statepoint file) and the number of mesh cells for
+        'mesh' domain types.
+    num_nuclides : int
+        The number of nuclides for which the multi-group cross section is
+        being tracked. This is unity if the by_nuclide attribute is False.
+    nuclides : Iterable of str or 'sum'
+        The optional user-specified nuclides for which to compute cross
+        sections (e.g., 'U238', 'O16'). If by_nuclide is True but nuclides
+        are not specified by the user, all nuclides in the spatial domain
+        are included. This attribute is 'sum' if by_nuclide is false.
+    sparse : bool
+        Whether or not the MGXS' tallies use SciPy's LIL sparse matrix format
+        for compressed data storage
+    loaded_sp : bool
+        Whether or not a statepoint file has been loaded with tally data
+    derived : bool
+        Whether or not the MGXS is merged from one or more other MGXS
+    hdf5_key : str
+        The key used to index multi-group cross sections in an HDF5 data store
+
+    """
+
+    def __init__(self, domain=None, domain_type=None,
+                 energy_groups=None, by_nuclide=False, name='', num_polar=1,
+                 num_azimuthal=1):
+        self._name = ''
+        self._rxn_type = None
+        self._by_nuclide = None
+        self._nuclides = None
+        self._estimator = 'tracklength'
+        self._domain = None
+        self._domain_type = None
+        self._energy_groups = None
+        self._num_polar = 1
+        self._num_azimuthal = 1
+        self._tally_trigger = None
+        self._tallies = None
+        self._rxn_rate_tally = None
+        self._xs_tally = None
+        self._sparse = False
+        self._loaded_sp = False
+        self._derived = False
+        self._hdf5_key = None
+        self._valid_estimators = ESTIMATOR_TYPES
+
+        self.name = name
+        self.by_nuclide = by_nuclide
+
+        if domain_type is not None:
```

Comment:
This check is not really needed. If `domain_type` is `None` it'll just remain `None`. This applies for the other checks below for `domain` and `energy_groups`.